design/project.go

```diff
@@ -0,0 +1,128 @@
+	Description("The project service performs operations on Projects.")
+
+	// Method describes a service method (endpoint)
+	Method("list", func() {
+		Description("Retrieve information about Projects registered in FuseML.")
+		Payload(func() {
+			// Field describes an object field given a field index, a field
+			// name, a type and a description.
+			Field(1, "name", String, "List only Project with given name", func() {
+				Example("mlflow-project-01")
+			})
+		})
+
+		Result(ArrayOf(Project), "Return all registered Projects.")
+
+		Error("NotFound", func() {
+			Description("If the Project is not found, should return 404 Not Found.")
+		})
+
+		HTTP(func() {
+			GET("/projects")
+			Param("name", String, "List only Project with given name", func() {
+				Example("mlflow-project-01")
+			})
+			Response(StatusOK)
+			Response("NotFound", StatusNotFound)
+		})
+
+		GRPC(func() {
+			Response(CodeOK)
+			Response("NotFound", CodeNotFound)
+		})
+
+	})
+
+	Method("get", func() {
+		Description("Retrieve a Project from FuseML.")
+
+		Payload(func() {
+			Field(1, "name", String, "Project name", func() {
+				Example("mlflow-project-01")
+			})
+			Required("name")
+		})
+
+		Error("BadRequest", func() {
+			Description("If name is not given, should return 400 Bad Request.")
+		})
+		Error("NotFound", func() {
+			Description("If there is no project with the given name, should return 404 Not Found.")
+		})
+
+		Result(Project)
+
+		HTTP(func() {
+			GET("/projects/{name}")
+			Response(StatusOK)
+			Response("BadRequest", StatusBadRequest)
+			Response("NotFound", StatusNotFound)
+		})
+
+		GRPC(func() {
+			Response(CodeOK)
+			Response("BadRequest", CodeInvalidArgument)
+			Response("NotFound", CodeNotFound)
+		})
+	})
+
+	Method("delete", func() {
+		Description("Delete a Project registered by FuseML.")
+
+		Payload(func() {
+			Field(1, "name", String, "Project name", func() {
+				Example("mlflow-project-01")
+			})
+			Required("name")
+		})
+
+		Error("BadRequest", func() {
+			Description("If name is not given, should return 400 Bad Request.")
+		})
+		Error("NotFound", func() {
+			Description("If there is no project with the given name, should return 404 Not Found.")
+		})
+
+		HTTP(func() {
+			DELETE("/projects/{name}")
+			Response(StatusNoContent)
+			Response("BadRequest", StatusBadRequest)
+		})
+		GRPC(func() {
+			Response(CodeOK)
+			Response("BadRequest", CodeInvalidArgument)
+		})
+	})
+})
+
+// Project describes the Project
+var Project = Type("Project", func() {
+
+	Field(1, "name", String, "The name of the Project", func() {
+		Example("mlflow-project-01")
+	})
+	Field(2, "users", ArrayOf(User), "Users assigned to the Project")
+	Field(3, "description", String, "Project description", func() {
+		Example("Set of MLFlow applications")
+		Default("")
+	})
+	Required("name")
+})
+
+// User describes the user assigned to the project
+var User = Type("User", func() {
+	Field(1, "name", String, "User name", func() {
```

Comment:
Is this the name of the user (Full Name) or the username?

Reply:
User name